mockgen/model/model.go

```diff
@@ -343,6 +343,13 @@ func typeFromType(t reflect.Type) (Type, error) {
 	}
 
 	if imp := t.PkgPath(); imp != "" {
+		// PkgPath might return a path that includes "vendor"
+		// These paths do not compile, so we need to remove everything
+		// up to and including "/vendor/"
+		// see https://github.com/golang/go/issues/12019
+		if i := strings.LastIndex(imp, "/vendor/"); i != -1 {
```

Comment:
`internal/` directories can only be accessed by packages in or descending from the directory immediately above the `internal/` directory.  It was a predecessor to the vendoring experiment started in Go 1.5.  Here's the details:

https://golang.org/doc/go1.4#internalpackages
https://docs.google.com/document/d/1e8kOo3r51b2BWtTs_1uADIA5djfXhPT36s6eHVRIvaU/edit?pref=2&pli=1

Mechanically, it works similar, except without the import path eliding that vendoring added.  Where package `github.com/foo`, when vendored, is imported as `github.com/foo`, if placed in the internal directory instead, it would have to be imported as, for example, `github.com/bar/internal/github.com/foo`.  However, like the vendored situation, it would only be accessible to that `bar` package and its subordinates.  Main difference is that instead of falling back to the $GOPATH/src location to check for an alternate (non-vendored) version, the internal one will throw a compile time error `use of internal package not allowed`.